cypress-shared/support/b2b/common.js

```diff
@@ -145,3 +145,48 @@ export function userAndCostCenterShouldNotBeEditable(
       .should('be.disabled')
   })
 }
+
+export function verifyImpersonationFeatureAvailable(
+  role,
+  user,
+  impersonation = false
+) {
+  it(`Verifying impersonate feature available for ${role}`, () => {
+    cy.gotoMyOrganization()
+    cy.get(selectors.PageBlock)
+      .eq(1)
+      .find(selectors.MyOrganizationUserContainer)
+      .then(($els) => {
+        let texts = Array.from($els, (el) => el.innerText)
+
+        texts = texts.splice(3, texts.length)
+        const indexOfUser = texts.indexOf(user)
+        const childIndex = indexOfUser + 2
+
+        cy.get(
+          `div[class=ReactVirtualized__Grid__innerScrollContainer] > div:nth-child(${childIndex}) > div`
+        )
+          .should('be.visible')
+          .click()
+        cy.get(selectors.ImpersonateButton)
```

Comment:
There is a function called  validateToastMsg from support/b2b/utils.js
Make use of it + add toast message text to validation_text.js file

eg:  validateToastMsg(TOAST_MSG.updated)